src/services/getEditsForFileRename.ts

```diff
@@ -44,8 +45,8 @@ namespace ts {
                 // If it resolved to something already, ignore.
                 if (checker.getSymbolAtLocation(importStringLiteral)) continue;
 
-                const resolved = program.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName);
-                if (resolved && contains(resolved.failedLookupLocations, oldFilePath)) {
+                const resolved = resolveModuleName(importStringLiteral.text, sourceFile.fileName, program.getCompilerOptions(), host as ModuleResolutionHost);

```

Comment:
But then using this is wrong if host provided custom resolution.... Because it completely ignore's host;s resolve module name.

Reply:
The problem is that `host.resolveModuleNames` only gives us the successfully resolved modules --  but for this feature we only care about resolutions that failed and need to know what modules it tried.

Reply:
I think you should get result from moduleCache if host.resolveModuleName is not present. In case host.ResolveModuleName is present, may be LanguageServiceHost has optional parameter that getResolvedModuleWithFailedLookupLocations which gives you resolution.

Reply:
I've added a host method for specializing failed lookup locations, but I think using the cache was a bad idea in the first place -- it leaves around memory that we won't need 99% of the time.